### ISSUE TITLE
How to stop timing transform from running again after re-render

### ISSUE THREAD

**Issue by pribeh** [software-mansion/react-native-reanimated#351]
I'm attempting to run an animation only _onPressTab(tab) but it re-triggers every time something re-renders the screen. What am I missing to ensure the animation doesn't run again if the screen is updated. The update which causes the animation to run again is triggered by a textinput with value={this.state.username.trim()}.

```
function runTiming(clock, value, dest) {
  const state = {
    finished: new Value(0),
    position: new Value(0),
    time: new Value(0),
    frameTime: new Value(0),
  };

  const config = {
    duration: 450,
    toValue: new Value(0),
    easing: Easing.inOut(Easing.cubic),
  };

  const timeSyncedWithClock = new Value(0); 

  return block([
	cond(
		clockRunning(clock),
		cond(eq(timeSyncedWithClock, 0), [
			set(state.time, clock),
			set(timeSyncedWithClock, 1),
		]),
		[
			set(timeSyncedWithClock, 0), 
			set(state.finished, 0),
			set(state.time, clock),
			set(state.position, value),
			set(state.frameTime, 0),
			set(config.toValue, dest),
			startClock(clock),
		],
	),
	timing(clock, state, config),
	cond(state.finished, stopClock(clock)),
	state.position,
  ]);
}

	constructor(props) {
		super(props);

		clock: new Clock(),
		state: new Value(0),
		trans: new Value(0),
	}

	_onPressTab(tab) {
		this.setState({
			tab: tab
		})
		this.trans = runTiming(new Clock(), 0, 24);
	}

        render() {
		<TouchableOpacity
			onPress={() => this._onPressTab(1)}
			disabled={this.state.tab === 1}
		>
		</TouchableOpacity>
		<TouchableOpacity
			onPress={() => this._onPressTab(2)}
			disabled={this.state.tab === 2}
		>
		</TouchableOpacity>
                 <Animated.Image style={[ {transform: [{ rotate: concat(this.trans, 'deg') }] }]} />
         }
```

**Comment by Elf2707** [comment]
Also curious how to do it in a right way. Right now I'm manage to do it this way:

```
runTiming(..., callback = () => {}) {
  cond(state.finished, [stopClock(clock), call([], callback)]);
}

...

_onPressTab(tab) {
  this.setState({
    tab: tab
  });

  this.trans = runTiming(new Clock(), 0, 24, () => {
    this.trans = runTiming(new Clock(), 24, 24);
  });
}
```

But it seems to me it's a hack at least it's looking like a hack. Would be great to have sort of `loop` property if it true rerun animation on each serenader otherwise run it only once and close in the final state.

**Comment by kmagiera** [comment]
In order to implement that you need to slightly change the way you think about defining this interaction.

With reanimated the best way to think of interactions is that you should be able to declare them at the component level and external events should not modify the interaction but only adjust the flow of already defined interaction.

What it means in this case is that you should have a timing animation defined before ahead in your component with a `dest` value stored in your component state. Then as a result of tab press you should modify the destination value (e.g. `this.state.dest.setValue(25)`) as opposed to instantiating a new interaction.

Hope the above helps. In case it doesn't feel free to open a new issue. It would be perfect if you could make an expo snack next time as it is way more easier for us to prototype some suggestions in snack instead of working on an incomplete react component code.
